File: 2996/smallest-missing-integer-greater-than-sequential-prefix-sum.c
int missingInteger(int* nums, int numsSize) {
    int prefix = nums[0]; // 一開始 prefix 先放第1個數
    for(int i=1; i<numsSize; i++){ //照題目，找合理的prefix sum
        if(nums[i-1]+1==nums[i]) prefix += nums[i];
        else break;
    }
    //因為只有50個數字
    int exist[53]={}; //做對照表，看有哪些數字存在
    for(int i=0; i<numsSize; i++) exist[nums[i]]=1;

    int ans = prefix; //開始逐項檢查
    while(ans<51 && exist[ans]==1){ //只要數字存在
        ans++; //就繼續查看下一個數
    }
    return ans;
}
//case 2/616: [29,30,31,32,33,34,35,36,37]
// 數字不能太大

File: 2996/smallest-missing-integer-greater-than-sequential-prefix-sum.py
# 有一堆連續的數字，要找到缺的數字
# 因為數字很少，只有50個數字，所以簡單。
class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        # 先找 連續數字的prefix sum
        prefix = nums[0] # 放 前 i 項的加總結果
        for i in range(1,len(nums)):
            if nums[i] == nums[i-1] + 1: # 還是連續的數字
                prefix += nums[i]
            else: # 如果沒有連續的數
                break # 就離開，再找下一個數字
        # 找下一個數字
        ans = prefix # 從 prefix sum 開始，逐一檢查
        while ans in nums: ans += 1
        return ans
